### DISCUSSION TITLE
Custom Node Run FUNCTION on every run not only on init

### DISCUSSION THREAD

**Discussion by rautex** [comfyanonymous/ComfyUI#1804] [Q&A]
Hi,
the Idea is feed some things by an api make  automatic some things

my code is working, i have only the problem that to generate is called only once at the start, is here a way to say that i want it on every run? 

`      
  import torch,urllib.request, json
    class SF2:
       
        def __init__(self):
            pass
        
        @classmethod
        def INPUT_TYPES(s):
            return {"required": { "dbid": ("INT", {"default": 0, "min": 0,  "step": 1}),
                                
                                
                                  }}
    
        RETURN_TYPES = ("LATENT", "STRING", "STRING",  "STRING",  "FLOAT","TEXT_TYPE",)
        RETURN_NAMES = ("Latent", "postiveMain", "postiveSupport",  "negative", "resize","APIDATA")
    
        FUNCTION = "generate"
        CATEGORY = "SF2"
    
        def generate(self, dbid):
            response = urllib.request.urlopen("api url") 
            jsonData = response.read();
            data = json.loads(jsonData)
            print("###################MYDATA#################")
            print(data)
            print(data['title'])
            latent = torch.zeros([1, 4, data['height'] // 8, data['width'] // 8])
            return ({"samples":latent}, data['title'], data['keywords'], "", float(data['resize']),jsonData,)
    
    
    # A dictionary that contains all nodes you want to export with their names
    # NOTE: names should be globally unique
    NODE_CLASS_MAPPINGS = {
        "SF2": SF2
    }
    
    # A dictionary that contains the friendly/humanly readable titles for the nodes
    NODE_DISPLAY_NAME_MAPPINGS = {
        "SF2": "SF2"
    }

`

**Comment by rautex** [comment]
i have found a workaround

add   "seed": ("INT", {"default": 0, "min": 0, "max": 0xffffffffffffffff} and then you can set it to random so the function will be called every time on a create image proc

**Comment by Siberpone** [comment]
Seconding this. I have a very similar problem: I'm working on a custom node that needs to be executed on every new generation, regardless of whether the input params changed or not. Funnily enough, I figured out the same workaround: just have a dummy INT input and connect an INT primitive to it and set it to random. I would also like to know if there's a "native" or, at least, more graceful way to force execute a node.
